lib/livebook/teams/app_deployment.ex

```diff
@@ -0,0 +1,100 @@
+defmodule Livebook.Teams.AppDeployment do
+  use Ecto.Schema
+  alias Livebook.FileSystem
+
+  @file_extension ".zip"
+
+  @type t :: %__MODULE__{
+          id: String.t() | nil,
+          filename: String.t() | nil,
+          slug: String.t() | nil,
+          sha: String.t() | nil,
+          title: String.t() | nil,
+          deployment_group_id: String.t() | nil,
+          file: FileSystem.File.t() | String.t() | nil,
+          deployed_by: String.t() | nil,
+          deployed_at: NaiveDateTime.t() | nil
+        }
+
+  @primary_key {:id, :string, autogenerate: false}
+  embedded_schema do
+    field :filename, :string
+    field :slug, :string
+    field :sha, :string
+    field :title, :string
+    field :deployment_group_id, :string
+    field :file, :map
+    field :deployed_by, :string
+
+    timestamps(updated_at: nil, inserted_at: :deployed_at)
+  end
+
+  @doc """
+  Creates a new app deployment from session with compressed file.
+  """
+  @spec new(Livebook.Session.Data.t()) :: {:ok, t()} | {:error, FileSystem.error()}
+  def new(%Livebook.Session.Data{} = data) do
```

Comment:
Ideally I would not pass the whole Session.Data here, only the parts that it needs: perhaps `data.notebook.file_entries` and `data.file`?

Reply:
I agree

Reply:
Actually, I'll pass the file path directly, so I wouldn't need to handle using `FileSystem.File`

Reply:
We need to handle `FileSystem.File` though, because the notebook may be persisted into an S3 file system (including all the attachments).

Reply:
Yes, we need file, the idea is that we will read its contents, zip everything, and then we should no longer need a reference to any file.